diff -ur ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/vc12win32/PhysX.vcxproj ./PhysX-3.4-master/PhysX_3.4/Source/compiler/vc12win32/PhysX.vcxproj
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/vc12win32/PhysX.vcxproj	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/vc12win32/PhysX.vcxproj	2017-08-25 18:40:17.377924700 +0300
@@ -86,8 +86,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationDEBUG_x86.dll /DELAYLOAD:PxPvdSDKDEBUG_x86.dll /DELAYLOAD:PhysX3CommonDEBUG_x86.dll /DEBUG</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;./../../../Lib/vc12win32/PhysX3GpuDEBUG_x86.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:PxFoundationDEBUG_x86.dll /DELAYLOAD:PxPvdSDKDEBUG_x86.dll /DELAYLOAD:PhysX3CommonDEBUG_x86.dll /DEBUG</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3DEBUG_x86.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -132,8 +132,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationCHECKED_x86.dll /DELAYLOAD:PxPvdSDKCHECKED_x86.dll /DELAYLOAD:PhysX3CommonCHECKED_x86.dll</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;./../../../Lib/vc12win32/PhysX3GpuCHECKED_x86.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:PxFoundationCHECKED_x86.dll /DELAYLOAD:PxPvdSDKCHECKED_x86.dll /DELAYLOAD:PhysX3CommonCHECKED_x86.dll</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3CHECKED_x86.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -178,8 +178,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationPROFILE_x86.dll /DELAYLOAD:PxPvdSDKPROFILE_x86.dll /DELAYLOAD:PhysX3CommonPROFILE_x86.dll /INCREMENTAL:NO /DEBUG</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;./../../../Lib/vc12win32/PhysX3GpuPROFILE_x86.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:PxFoundationPROFILE_x86.dll /DELAYLOAD:PxPvdSDKPROFILE_x86.dll /DELAYLOAD:PhysX3CommonPROFILE_x86.dll /INCREMENTAL:NO /DEBUG</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3PROFILE_x86.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -224,8 +224,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundation_x86.dll /DELAYLOAD:PxPvdSDK_x86.dll /DELAYLOAD:PhysX3Common_x86.dll /INCREMENTAL:NO</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;./../../../Lib/vc12win32/PhysX3Gpu_x86.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:PxFoundation_x86.dll /DELAYLOAD:PxPvdSDK_x86.dll /DELAYLOAD:PhysX3Common_x86.dll /INCREMENTAL:NO</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3_x86.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
diff -ur ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/vc12win64/PhysX.vcxproj ./PhysX-3.4-master/PhysX_3.4/Source/compiler/vc12win64/PhysX.vcxproj
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/vc12win64/PhysX.vcxproj	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/vc12win64/PhysX.vcxproj	2017-08-25 18:40:41.234138700 +0300
@@ -85,8 +85,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationDEBUG_x64.dll /DELAYLOAD:PxPvdSDKDEBUG_x64.dll /DELAYLOAD:PhysX3CommonDEBUG_x64.dll /DEBUG</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;./../../../Lib/vc12win64/PhysX3GpuDEBUG_x64.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:PxFoundationDEBUG_x64.dll /DELAYLOAD:PxPvdSDKDEBUG_x64.dll /DELAYLOAD:PhysX3CommonDEBUG_x64.dll /DEBUG</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3DEBUG_x64.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -130,8 +130,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationCHECKED_x64.dll /DELAYLOAD:PxPvdSDKCHECKED_x64.dll /DELAYLOAD:PhysX3CommonCHECKED_x64.dll</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;./../../../Lib/vc12win64/PhysX3GpuCHECKED_x64.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:PxFoundationCHECKED_x64.dll /DELAYLOAD:PxPvdSDKCHECKED_x64.dll /DELAYLOAD:PhysX3CommonCHECKED_x64.dll</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3CHECKED_x64.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -175,8 +175,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationPROFILE_x64.dll /DELAYLOAD:PxPvdSDKPROFILE_x64.dll /DELAYLOAD:PhysX3CommonPROFILE_x64.dll /INCREMENTAL:NO /DEBUG</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;./../../../Lib/vc12win64/PhysX3GpuPROFILE_x64.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:PxFoundationPROFILE_x64.dll /DELAYLOAD:PxPvdSDKPROFILE_x64.dll /DELAYLOAD:PhysX3CommonPROFILE_x64.dll /INCREMENTAL:NO /DEBUG</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3PROFILE_x64.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -220,8 +220,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundation_x64.dll /DELAYLOAD:PxPvdSDK_x64.dll /DELAYLOAD:PhysX3Common_x64.dll /INCREMENTAL:NO</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;./../../../Lib/vc12win64/PhysX3Gpu_x64.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:PxFoundation_x64.dll /DELAYLOAD:PxPvdSDK_x64.dll /DELAYLOAD:PhysX3Common_x64.dll /INCREMENTAL:NO</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3_x64.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
diff -ur ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/vc14win32/PhysX.vcxproj ./PhysX-3.4-master/PhysX_3.4/Source/compiler/vc14win32/PhysX.vcxproj
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/vc14win32/PhysX.vcxproj	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/vc14win32/PhysX.vcxproj	2017-08-25 18:41:03.561255800 +0300
@@ -86,8 +86,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationDEBUG_x86.dll /DELAYLOAD:PxPvdSDKDEBUG_x86.dll /DELAYLOAD:PhysX3CommonDEBUG_x86.dll /DEBUG</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;./../../../Lib/vc14win32/PhysX3GpuDEBUG_x86.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:PxFoundationDEBUG_x86.dll /DELAYLOAD:PxPvdSDKDEBUG_x86.dll /DELAYLOAD:PhysX3CommonDEBUG_x86.dll /DEBUG</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3DEBUG_x86.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -132,8 +132,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationCHECKED_x86.dll /DELAYLOAD:PxPvdSDKCHECKED_x86.dll /DELAYLOAD:PhysX3CommonCHECKED_x86.dll</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;./../../../Lib/vc14win32/PhysX3GpuCHECKED_x86.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:PxFoundationCHECKED_x86.dll /DELAYLOAD:PxPvdSDKCHECKED_x86.dll /DELAYLOAD:PhysX3CommonCHECKED_x86.dll</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3CHECKED_x86.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -178,8 +178,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationPROFILE_x86.dll /DELAYLOAD:PxPvdSDKPROFILE_x86.dll /DELAYLOAD:PhysX3CommonPROFILE_x86.dll /INCREMENTAL:NO /DEBUG</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;./../../../Lib/vc14win32/PhysX3GpuPROFILE_x86.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:PxFoundationPROFILE_x86.dll /DELAYLOAD:PxPvdSDKPROFILE_x86.dll /DELAYLOAD:PhysX3CommonPROFILE_x86.dll /INCREMENTAL:NO /DEBUG</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3PROFILE_x86.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -224,8 +224,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundation_x86.dll /DELAYLOAD:PxPvdSDK_x86.dll /DELAYLOAD:PhysX3Common_x86.dll /INCREMENTAL:NO</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;./../../../Lib/vc14win32/PhysX3Gpu_x86.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x86 /DELAYLOAD:PxFoundation_x86.dll /DELAYLOAD:PxPvdSDK_x86.dll /DELAYLOAD:PhysX3Common_x86.dll /INCREMENTAL:NO</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/Win32/nvToolsExt32_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3_x86.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
diff -ur ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/vc14win64/PhysX.vcxproj ./PhysX-3.4-master/PhysX_3.4/Source/compiler/vc14win64/PhysX.vcxproj
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/vc14win64/PhysX.vcxproj	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/vc14win64/PhysX.vcxproj	2017-08-25 18:41:24.353741300 +0300
@@ -85,8 +85,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationDEBUG_x64.dll /DELAYLOAD:PxPvdSDKDEBUG_x64.dll /DELAYLOAD:PhysX3CommonDEBUG_x64.dll /DEBUG</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;./../../../Lib/vc14win64/PhysX3GpuDEBUG_x64.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:PxFoundationDEBUG_x64.dll /DELAYLOAD:PxPvdSDKDEBUG_x64.dll /DELAYLOAD:PhysX3CommonDEBUG_x64.dll /DEBUG</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3DEBUG_x64.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -130,8 +130,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationCHECKED_x64.dll /DELAYLOAD:PxPvdSDKCHECKED_x64.dll /DELAYLOAD:PhysX3CommonCHECKED_x64.dll</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;./../../../Lib/vc14win64/PhysX3GpuCHECKED_x64.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:PxFoundationCHECKED_x64.dll /DELAYLOAD:PxPvdSDKCHECKED_x64.dll /DELAYLOAD:PhysX3CommonCHECKED_x64.dll</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3CHECKED_x64.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -175,8 +175,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundationPROFILE_x64.dll /DELAYLOAD:PxPvdSDKPROFILE_x64.dll /DELAYLOAD:PhysX3CommonPROFILE_x64.dll /INCREMENTAL:NO /DEBUG</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;./../../../Lib/vc14win64/PhysX3GpuPROFILE_x64.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:PxFoundationPROFILE_x64.dll /DELAYLOAD:PxPvdSDKPROFILE_x64.dll /DELAYLOAD:PhysX3CommonPROFILE_x64.dll /INCREMENTAL:NO /DEBUG</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3PROFILE_x64.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
@@ -220,8 +220,8 @@
 			<DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
 		</ClCompile>
 		<Link>
-			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:nvcuda.dll /DELAYLOAD:PxFoundation_x64.dll /DELAYLOAD:PxPvdSDK_x64.dll /DELAYLOAD:PhysX3Common_x64.dll /INCREMENTAL:NO</AdditionalOptions>
-			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;./../../../Lib/vc14win64/PhysX3Gpu_x64.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalOptions>/MAP /MACHINE:x64 /DELAYLOAD:PxFoundation_x64.dll /DELAYLOAD:PxPvdSDK_x64.dll /DELAYLOAD:PhysX3Common_x64.dll /INCREMENTAL:NO</AdditionalOptions>
+			<AdditionalDependencies>./../../../../Externals/nvToolsExt/1/lib/x64/nvToolsExt64_1.lib;%(AdditionalDependencies)</AdditionalDependencies>
 			<OutputFile>$(OutDir)PhysX3_x64.dll</OutputFile>
 			<AdditionalLibraryDirectories>%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
 			<ProgramDatabaseFile>$(TargetDir)\$(TargetName).pdb</ProgramDatabaseFile>
